styling updates

michellexliu opened the pull request:
* Added _ to .scss file names
* Re-factored styling_classnames file into @extend in sass
* Updated spacing (decreased padding between buttons by 1 token, fixed focus state margin issues, decreased margins between checkboxes by 1 token, smaller line spacing for title, more space between header & body text)
* Added drop shadow on module

michellexliu committed:
added _ to scss file names, started testing @extend

michellexliu committed:
styling changes for checkboxes

michellexliu committed:
Merge branch 'dev' into story/sass-refactor

michellexliu committed:
converted rest of styling-classnames to @extend, added drop shadow on module, fixed focus state margin issue

michellexliu committed:
updated spacing in plaintext section, turned plaintext into div and set innerhtml to p in order to control spacing in between paragraphs

ksiegel1923 commented on feedback-module/src/translations/en/translation.js:
```diff
@@ -65,15 +64,15 @@ export const TRANSLATIONS_EN = {
       "Thank you for your feedback! It has been submitted.<br /><br />Your feedback is anonymous and confidential, so you will not receive a reply.",
     plainText:
-      "The City of New York is always trying to improve its services. Are you interesed in being a user research participant? This is is entirely voluntary and you can opt out at any time. Signing up to be a user research participant will have no impact on your feedback today, or your eligibility to access or receive services in the future.",
+      "<p>The City of New York is always trying to improve its services. Are you interesed in being a user research participant? This is is entirely voluntary and you can opt out at any time. Signing up to be a user research participant will have no impact on your feedback today, or your eligibility to access or receive services in the future.</p>",
```
Are we sure we want paragraph tags inside the translation files? I don't know if that would be considered best practice because it could interfere with the actual translation that goes on

michellexliu replied:
Yeah, I thought about this too, but I'm just not sure how else to customize the spacing between each paragraph of the body text (since design pointed out that there should be more spacing between each paragraph, but still keep it less than the spacing between the title and paragraph). I had considered just adding in another `<br />`, but we can't really customize how large that space is. I think another option would be to make the `plainText` an array, and then in `Module.js` map each piece of text into a `<p>` tag with our desired padding— not sure if this would be over-complicating things for our current purposes; it could become necessary if we ever end up having even more paragraphs per section in the future though. What do you think?

ksiegel1923 replied:
I think we may want to find another solution. Additionally I don't know if we necessarily want <br /> within our translations files either. Maybe something we should ask Rapi about.

michellexliu replied:
Yeah, I definitely agree that we should ask Rapi about this. Which part are you referring to in your second sentence?

ksiegel1923 replied:
Sorry I meant line breaks within the files either

ksiegel1923 commented on feedback-module/src/components/Module.js:
```diff
@@ -161,37 +149,35 @@ function Module({ pagetitle, endpoint, dir }) {
+          <Grid className={"bg-primary feedback-screen-container"}>
             <p
-              className={`${H1_WHITE_STYLE} ${dir === "rtl" && "text-right"}`}
+              className="font-sans-md2 feedback-h1 feedback-h1--light"
```
Do we need both "feedback-h1 and feedback-h1--light"? Isn't the purpose of using extends for --light so that it automatically inherits the styles from feedback-h1?

michellexliu replied:
I'd considered just using `feedback-h1--light`, but according to the official BEM docs, we should keep in the original class and add the new one separately (see http://getbem.com/naming/ Modifier section).

ksiegel1923 commented on feedback-module/src/styles/_components.scss:
```diff
@@ -0,0 +1,136 @@
+  p {
+    @extend .margin-0;
+  }
+}
```
If two classes share a lot of styling in common can one of them extend the other so we don't have so much repetition within this file?

michellexliu replied:
Hmmm the thing with that is that if we have one extend the other it would extend _all_ of the styling from the other (if we're also considering the uswds classes, it'd also extend the default uswds styling for one of those classes as well, that we didn't set). I think maybe an option to handle the common styling would be to use grouping selectors and move all the common styles in there?

For example
```
.usa-label, .usa-checkbox, .feedback-h1, etc {
@extend .text-bold, .text-primary;
}
```

What do you think?

ksiegel1923 replied:
Yes I think that makes sense, I think we should do that

michellexliu committed:
grouped commmon styles together, created mixin to set mobile-lg styles in order to avoid setting mobile-lg- and mobile-lg\: repeatedly

ksiegel1923 committed:
header logo resolution and alignment:
changed header logo from a png to an svg (help with resolution)
added extra margin and padding to header logo so visually it appears centered

michellexliu committed:
re-organized sass architecture: created abstracts folder, set up themes folder in case we want to support multiple themes for stretch goal, split components into different files for improved modularity

michellexliu committed:
renamed feedback-module classes to follow bem conventions

michellexliu committed:
Merge branch 'dev' into story/sass-refactor

michellexliu committed:
updated config for plainText to array

michellexliu committed:
deleted console.log

michellexliu committed:
deleted unnecessary div

michellexliu committed:
Merge branch 'story/header-logo' of https://github.com/nyc-cto/feedback-module into story/header-logo

michellexliu committed:
added header changes

michellexliu committed:
changed from last-child to last-of-type

michellexliu committed:
updated checkbox size, moved logo styling to .scss file

michellexliu committed:
set buttons to be full width on mobile view

michellexliu committed:
implemented rapi's suggestions: renamed h1 to heading, deleted _ from export.module.scss name

michellexliu committed:
fixed export.module.scss variables import
